File: flutter_inappwebview_forge_android/android/src/main/kotlin/com/emirkanacar/flutter_inappwebview_forge_android/types/DownloadStartResponse.kt
package com.emirkanacar.flutter_inappwebview_forge_android.types

open class DownloadStartResponse(
    open var handled: Boolean,
    open var action: Int?,
    open var resultFilePath: String?
) {
    companion object {
        @JvmStatic
        fun fromMap(map: MutableMap<String, Any?>?): DownloadStartResponse? {
            if (map == null) return null
            val handled = map["handled"] as? Boolean ?: false
            val action = (map["action"] as? Number)?.toInt()
            val resultFilePath = map["resultFilePath"] as? String
            return DownloadStartResponse(handled, action, resultFilePath)
        }
    }
}

File: flutter_inappwebview_forge_macos/macos/flutter_inappwebview_forge_macos/Sources/flutter_inappwebview_forge_macos/Types/DownloadStartResponse.swift
import Foundation

public class DownloadStartResponse: NSObject {
    var handled: Bool
    var action: Int?
    var resultFilePath: String?

    public init(handled: Bool, action: Int?, resultFilePath: String?) {
        self.handled = handled
        self.action = action
        self.resultFilePath = resultFilePath
    }

    public static func fromMap(map: [String: Any?]?) -> DownloadStartResponse? {
        guard let map = map else { return nil }
        return DownloadStartResponse(
            handled: map["handled"] as? Bool ?? false,
            action: map["action"] as? Int,
            resultFilePath: map["resultFilePath"] as? String
        )
    }
}
